pkg/endpointmanager/host.go

```diff
@@ -47,6 +48,13 @@ func (mgr *endpointManager) updateHostEndpointLabels(oldNodeLabels, newNodeLabel
 		return
 	}
 
+	newNodeIdtyLabels, _ := labelsfilter.Filter(labels.Map2Labels(newNodeLabels, labels.LabelSourceK8s))
```

Comment:
One thing I'm wondering about - the only usage of `updateHostEndpointLabels` is by the node labels observer above.   When a local node change is observed this does:

https://github.com/cilium/cilium/blob/d7a851ddc0744a87c001aa35b5af2d84b3e41b9a/pkg/endpointmanager/host.go#L35-L37

So this will compare all labels, and return before calling updateHostEndpointLabels.  Could we not just compare the relevant identity labels once?